packwerk.gemspec

```diff
@@ -50,6 +50,7 @@ Gem::Specification.new do |spec|
   # For Ruby parsing
   spec.add_dependency("ast")
   spec.add_dependency("parser")
+  spec.add_dependency("prism", ">= 0.22.0") # 0.21.0 fixes numbered block params, 0.22.0 adds Ruby 2.7 compatibility
```

Comment:
Do we need to use parser still or can we drop it as a dependency now that we're on prism?

Reply:
Yeah besides, in this PR packwerk still uses `Parser::AST` and a few other things.

A "clean" migration to Prism without using the translation layer, and using prism's AST format instead, would let us get rid of `parser` AFAICT. But it's significantly more work.

I've looked into it a bit, and among other things `NodeHelpers` has to be completely rewritten. I'm thinking we may want to completely remove it though since prism's AST format is a lot more user friendly and may not need this kind of leaky wrapper around it.

(@kddnewton the only thing that's been a little annoying is that the types don't seem to represent shared interfaces - e.g. between symbol and string nodes, or between class and module nodes. What's a good place to discuss these things?)

Reply:
Looks like on a small app, we get around the same to slightly better performance that takes roughly `0.9s` to complete:
Before:
```
📦 Finished in 0.9 seconds
```
After:
```
📦 Finished in 0.78 seconds
```

 On a larger app, we see a significant slowdown for cached runs:
Before:
```
📦 Finished in 18.31 seconds
```
After:
```
📦 Finished in 30.77 seconds
```

I can't merge this until we find out why this is happening.

Reply:
Before:
```
📦 Finished in 16.88 seconds

bin/packwerk check  0.30s user 0.04s system 1% cpu 33.053 total
```
After:
```
📦 Finished in 17.98 seconds

bin/packwerk check  0.29s user 0.04s system 0% cpu 34.413 total
```

Looks like we're at the same-ish difference to me now. How many times faster do we expect Prism to be using the translation layer?

Reply:
@exterm do you have an application that benefits from this change? If you do, please provide benchmarks.

Reply:
@kddnewton parsing is about 2/3 of packwerk runtime on large apps. So I'd expect a big speedup and I did get a big speedup on the app I was working on (~1 million Ruby LOC). I'll do another benchmark after updating to prism 0.23.0